### ISSUE TITLE
Make zaptec an official HA integration?

### ISSUE THREAD

**Issue by sveinse** [custom-components/zaptec#129]
Currently the zaptec integration is distributed via the HACS framework.

Would it be interesting to have zaptec as an official HA integration? I'd like to hear from the people that are using the integration. Migrating the zaptec integration from a HACS installable integration into a official HA integration will require considerable efforts. There is no point in doing it unless there is a need and would benefit our users.

What do you think? Is using HACS for this repo a burden for making use of zaptec?

**Comment by c0mplex1** [comment]
For me personally it is not really an extra burden to use Zaptec integration via HACS, but I think that if the integration becomes part of HA OS it will work even more reliably than it is now. After all, your integration does not need HACS anymore.

**Comment by dsmagghe** [comment]
would be nice, but even nicer if the load balancer would also be able to balance multiple chargers at once. I see the last commit there was more then 9 months ago :(

**Comment by Ghannes** [comment]
> Currently the zaptec integration is distributed via the HACS framework.
> 
> Would it be interesting to have zaptec as an official HA integration? I'd like to hear from the people that are using the integration. Migrating the zaptec integration from a HACS installable integration into a official HA integration will require considerable efforts. There is no point in doing it unless there is a need and would benefit our users.
> 
> What do you think? Is using HACS for this repo a burden for making use of zaptec?

What are the addiotional efforts?

**Comment by sveinse** [comment]
> What are the addiotional efforts?

- The Zaptec api itself must be separated out to a separate package on pypi
- The coding standard must be updated to fulfil the [Home Assistant quality scale](https://developers.home-assistant.io/docs/integration_quality_scale_index/) 
- A series of PRs adding the Zaptec integration into ha-core. Here there is a convention to adding only one feature or file at a time,  in order to make the changes small enough to be reviewable. So there will be many PRs and it will be waiting on collaborators able to approve.

More info here: https://developers.home-assistant.io/docs/creating_component_code_review

**Comment by Hellowlol** [comment]
It really not worth it imo

**Comment by c0mplex1** [comment]
@sveinse 
Considering the huge amount of work that needs to be put into making the zaptec integration part of HA, I now also wonder if it is worth it.
Will the adjustment and/or expansion of the integration remain as flexible as it is now, or will it become a sluggish process if it has to be approved by many collaborators first.
Decisions, decisions, decisions.

**Comment by pgrodem** [comment]
I would think that making it an official integration would make it accessible to more users, both from a discoverability standpoint, but also with trust or status, or whatever you want to call it. It seems to require quite some work, but wouldn't that sort of standardization also make it easier for others to contribute down the road?

**Comment by sveinse** [comment]
I think the conclusion is not to go for HA integration for now. We need a more active community to be able to drive such an effort. Closing this issue.

**Comment by steinmn** [comment]
While I'm not in a position to commit to any kind of timeline, I do think of a core integration as a logical goal to be working towards. Thinking of starting PRs to core seems extremely premature at the moment, but to keep HA quality standards and principles as the basis for design choices and prioritization as I think (hope?) we have been doing lately is a direction I like, and one I would like to continue to contribute to.

What kind of contact have you had with the HA foundation? Would it make sense to ask them for some high level pointers on what topics we should address that aren't already covered by the [issue list](https://github.com/custom-components/zaptec/issues)? And/or some tips to kickstart the test-setup we know is probably our biggest deficiency at the moment.

**Comment by sveinse** [comment]
Starting a PR for core integration _right now_ was definitely not on my mind. If we do plan to do so, there are a lot of things we have to do to get there first. Those things happen to be aligned with the journey that we're on, so its not like its a completely different direction for us. It just sets the long term aspirations.

One such thing that comes with migration to core is that the API interface parts must be siphoned off to a separate and official package on pypi. I have been toying with the idea to start that work. We could use this even without deciding to migrate to core. It has some pros and some cons.

As for the contact from HA foundation: This was about how to discuss how to deal with the policy from Zaptec vs the policy from HA,  with things like that poll intervals and the special coordinators. I got some guidance on how to set this up in a HA-compliant-like manner. I believe the biggest upside of being included into core, is the access to all the people that knows HA very well.

I have, and I'm sure you have too, spent a lot of time figuring out how things are wired together in HA. Since our community is small, I've wished many times for having access to more eyes to our PRs. Especially from the HA world, which is so complex and ever evolving. I've dreamt about it existing some kind of HA-service for reviews. I think that's what you get from being in core.

...or perhaps this is a pipe dream. Being in core is not as collaborative as I think it is. That is still puts the same weight on the maintainers as an independent integration.
